### DISCUSSION TITLE
Implement pagination without tokens

### DISCUSSION THREAD

**Discussion by Padeanu** [stac-utils/stac-fastapi-pgstac#167] [Q&A]
Hello, maybe i miss this while reading the documentation, but is there a way to implement basic pagination, without tokens?

https://github.com/radiantearth/stac-api-spec/tree/release/v1.0.0/item-search#pagination
I manually created the links, but stac_fastapi_pgstac modifies them, by adding token=next:*

Based on given input url, the next and prev links should be url?page=x+1 or url?page=x-1

**Comment by vincentsarago** [comment]
stac specs support multi pagination (https://github.com/stac-utils/stac-fastapi/tree/main/stac_fastapi/extensions/stac_fastapi/extensions/core/pagination) schemas but pgstac won't support `page=`, it will only support `token=` 

The pagination is currently broken (on pgstac) side and we are waiting on a pgstac update in order to finish some pagination PR on stac-fastapi-pgstac

**Reply by Padeanu** [reply]
Ok, thank you. I will periodically keep an eye on pull requests for that change.
